python/ray/data/_internal/planner/map_rows.py

```diff
@@ -21,9 +22,13 @@ def fn(
         for block in blocks:
             block = BlockAccessor.for_block(block)
             for row in block.iter_rows():
-                output_buffer.add(row_fn(row))
-                if output_buffer.has_next():
-                    yield output_buffer.next()
+                row = row_fn(row)
```

Comment:
Hmm why do we need this support for map_rows? Shouldn't it be for map_batches and flat_map only?

Reply:
right, updated